src/firebase/auth.ts

```diff
@@ -141,4 +144,13 @@ export const signUserOut = async (dispatch: AuthDispatch) => {
     console.warn('Sign out error', error);
   }
   dispatch({ type: 'SIGN_OUT' });
-};
\ No newline at end of file
+};
+
+export const updateLanguage = async (
+  dispatch: AuthDispatch,
+  langUpdate: React.Dispatch<React.SetStateAction<Dictionary>>,
+  params: { language: string },
+) => {
+  langUpdate(checkAndGetLang(params.language));
```

Comment:
when calling this function make sure to also run a firebase query to change the language preference for the user document.